tests/integration/Storefront/Controller/CheckoutControllerTest.php

```diff
@@ -456,8 +453,9 @@ public function testCheckoutConfirmPageLoadedHookScriptsAreExecuted(): void
         $contextToken = Uuid::randomHex();
 
         $cart = $this->fillCart($contextToken);
-
         $salesChannelContext = $this->createSalesChannelContext($contextToken, $cart->getTransactions()->first()?->getPaymentMethodId());
+        static::getContainer()->get(CartPersister::class)->save($cart, $salesChannelContext);
```

Comment:
Is that related? Just because I didn't see any mention of that in the description

Reply:
no, the test was failing because the cart was created but not persisted. but the [CheckoutConfirmPageLoader](https://github.com/shopware/shopware/blob/trunk/src/Storefront/Page/Checkout/Confirm/CheckoutConfirmPageLoader.php#L59) loads the cart without caching. so fresh from the DB. but the before added lineItems were not persisted, so the cart was empty and that let the test fail